network-config-analyzer/CanonicalIntervalSet.py

```diff
@@ -214,9 +214,9 @@ def touches(self, other):
             :return: Whether the two intervals touch each other (without overlapping)
             :rtype: bool
             """
-            if self.start > other.end:
+            if self.start > other.end and (not hasattr(other.end, 'max_address') or not callable(other.end.max_address) or not other.end.max_address()):
                 return self.start == other.end + 1
-            if other.start > self.end:
+            if other.start > self.end and (not hasattr(self.end, 'max_address') or not callable(self.end.max_address) or not self.end.max_address()):
```

Comment:
Let's leave this code as is, and instead catch the exception in IPNetworkAddress.__add__()